modules/sdk-coin-avaxp/src/lib/transaction.ts

```diff
@@ -258,16 +261,57 @@ export class Transaction extends BaseTransaction {
   /** @inheritdoc */
   explainTransaction(): TransactionExplanation {
     const txJson = this.toJson();
-    const displayOrder = ['id', 'outputAmount', 'changeAmount', 'outputs', 'changeOutputs', 'fee', 'type'];
+
+    let totalIn = new BN(0);
+    txJson.utxos.map((utxo) => {
+      totalIn = totalIn.add(new BN(utxo.amount));
+    });
+    const totalStaked = txJson.outputs.reduce((p, n) => p.add(new BN(n.value)), new BN(0));
+    const totalOut = totalStaked;
+    const totalChange = totalIn.sub(totalOut);
+    const rewardAddresses = this.rewardAddresses;
+    const nodeID =
+      this.type === TransactionType.addValidator
+        ? (this.avaxPTransaction as AddValidatorTx).getNodeIDString()
+        : undefined;
```

Comment:
do we need this business logic in `explainTransaction`? can it be moved to `.toJson` instead?